pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/admin/VipStatus.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.proxy.server.admin;
+
+import java.io.File;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response.Status;
+
+/**
+ * Web resource used by the VIP service to check to availability of the proxy instance.
+ */
+@Path("/status.html")
+public class VipStatus extends ProxyWebResource {
+
+    @GET
+    @Context
+    public String checkStatus() {
+        String statusFilePath = service().getConfiguration().getStatusFilePath();
+
+        File statusFile = new File(statusFilePath);
+        if (statusFile.exists()) {
+            return "OK";
+        } else {
+            throw new WebApplicationException(Status.NOT_FOUND);
```

Comment:
Is it possible to return 404 without generating an exception? Or do you want to generate an exception for some reasons?

Reply:
OK, we shouldn't "break" the pattern. Let's keep it for now.
It seems that there is a mechanism called `ExceptionMapper` and it sets status code during exception handling, it sounds like too costly for status.html though.